### PR TITLE
Add default converter option in withConverter()

### PR DESCRIPTION
**Pull request by thebrianchen**
Porting from [node](https://github.com/googleapis/nodejs-firestore/pull/1394).

### CODE REVIEW

File: packages/firestore/src/api/database.ts [hunk 1]
```diff
@@ -830,14 +830,20 @@ export class DocumentReference<T = PublicDocumentData>
...
 
+  withConverter(converter: null): PublicDocumentReference<PublicDocumentData>;
   withConverter<U>(
     converter: PublicFirestoreDataConverter<U>
-  ): PublicDocumentReference<U> {
-    return new DocumentReference<U>(
+  ): PublicDocumentReference<U>;
+  withConverter<U>(
+    converter: PublicFirestoreDataConverter<U> | null
+  ): PublicDocumentReference<U> | PublicDocumentReference<PublicDocumentData> {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/api/database.ts:839]: In Node, this is just `PublicDocumentReference<U>` and TypeScript deduces U.

File: packages/firestore/src/api/database.ts [hunk 2]
```diff
@@ -1141,12 +1147,18 @@ export class Query<T = PublicDocumentData>
     return onSnapshot(this._delegate, options, observer);
   }
 
-  withConverter<U>(converter: PublicFirestoreDataConverter<U>): Query<U> {
-    return new Query<U>(
+  withConverter(converter: null): Query<PublicDocumentData>;
+  withConverter<U>(converter: PublicFirestoreDataConverter<U>): Query<U>;
+  withConverter<U>(
+    converter: PublicFirestoreDataConverter<U> | null
+  ): Query<U> | Query<DocumentData> {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/api/database.ts:1154]: Same as above.

File: packages/firestore/src/api/database.ts [hunk 3]
```diff
@@ -1283,14 +1295,20 @@ export class CollectionReference<T = PublicDocumentData>
...
 
+  withConverter(converter: null): CollectionReference<PublicDocumentData>;
   withConverter<U>(
     converter: PublicFirestoreDataConverter<U>
-  ): CollectionReference<U> {
-    return new CollectionReference<U>(
+  ): CollectionReference<U>;
+  withConverter<U>(
+    converter: PublicFirestoreDataConverter<U> | null
+  ): CollectionReference<U> | CollectionReference<PublicDocumentData> {
```
Comment (schmidt-sebastian) [inline packages/firestore/src/api/database.ts:1304]: And one more for old times sake.

**Review by schmidt-sebastian — APPROVED** [review]
Thanks!

**Review by Feiyang1 — APPROVED** [review]

**Pull request merged**